src/tests/modify.test.js

```diff
@@ -0,0 +1,661 @@
+            required: 'We need to know your pet name.',
+          },
+        },
+        pet_address: {
+          properties: {
+            street: {
+              'x-jsf-errorMessage': {
+                required: 'Your pet cannot live in the street.',
+              },
+            },
+          },
+        },
+      },
+    });
+  });
+
+  it('reorder fields', () => {
+    const baseExample = {
+      properties: {
+        /*...*/
+      },
+      'x-jsf-order': ['field_a', 'field_b', 'field_c', 'field_d'],
+    };
+    const result = modify(baseExample, {
+      // eslint-disable-next-line no-unused-vars
+      order: (originalOrder) => {
+        // [*4]
+        // console.log(order) // ['field_a', 'field_b', 'field_c', 'field_d']
+        return {
+          order: ['field_c', 'field_a', 'field_b'],
+          rest: 'start', // 'start' | 'end'
+        };
+      },
+    });
+
+    // 💡 Note how the missing field (field_d) got automatically added as safety measure.
+    expect(result).toMatchObject({
+      'x-jsf-order': ['field_d', 'field_c', 'field_a', 'field_b'],
+    });
+  });
+
+  it.todo('reorder fields in fieldsets');
+
+  // Q:❓ Would this be useful for first version or too advanced?
+  it.skip('customize conditional effects', () => {
+    const result = modify(schemaPet, {
+      conditionals: {
+        // The idea is to target a conditional by its id. (new!)
+        pet_conditional_id: {
+          then: {
+            has_pet: {
+              description: 'Glad you have a pet!',
+            },
+          },
+        },
+      },
+    });
+
+    expect(result).toMatchObject({
+      allOf: [
+        {
+          id: 'pet_conditional_id',
+          if: {
+            /*...*/
+          },
+          then: {
+            /*...*/
+          },
+          else: {
+            properties: {
+              has_pet: {
+                description: 'You should think about finding one.',
+              },
+            },
+          },
+        },
+      ],
+    });
+  });
+});
+
+const schemaTickets = {
+  properties: {
+    age: {
+      title: 'Age',
+      type: 'integer',
+    },
+    quantity: {
+      title: 'Quantity',
+      type: 'integer',
+    },
+    has_premium: {
+      title: 'Premium',
+      type: 'string',
+    },
+    premium_id: {
+      title: 'Premium ID',
+      type: 'boolean',
+    },
+  },
+  'x-jsf-order': ['age', 'quantity', 'has_premium', 'premium_id'],
+  allOf: [
+    {
+      if: {
+        properties: {
+          has_premium: {
+            const: 'yes',
+          },
+        },
+        required: ['has_premium'],
+      },
+      then: {
+        required: ['premium_id'],
+      },
+      else: {
+        properties: {
+          premium_id: false,
+        },
+      },
+    },
+  ],
+};
+
+describe('modify() - structures', () => {
+  // [*6]
+  it('pick fields - basic', () => {
+    const result = modify(schemaTickets, {
+      pick: {
+        fields: ['quantity'],
```

Comment:
What would happen if we picked a field that depend on other fields validations or conditions?